assets/js/theme/global/modal.js

```diff
@@ -196,12 +213,89 @@ export class Modal {
         this.$content.html('');
     }
 
+    setupFocusableElements(modalType) {
+        this.$preModalFocusedEl = $(document.activeElement);
+        
+        const $collection = focusableElements[modalType]();
+
+        this.findElementToFocus($collection);
+
+        const onTabbingWithArgs = (event) => this.onTabbing(event, modalType, this.findElementToFocus.bind(this));
+
+        $('#modal').on('keydown', onTabbingWithArgs);
+    }
+
+    findElementToFocus($candidatesCollection, $reserveCollection = null) {
+        $candidatesCollection.each((index, element) => {
+            const $focusCanditate = $(element);
+            $focusCanditate.focus();
+            if ($focusCanditate.is($(document.activeElement))) {
+                return false;
+            };
+            // if no appropriate candidate - find appropriate in full collection
+            if (index === $candidatesCollection.length - 1 && $reserveCollection) {
+                this.findElementToFocus($reserveCollection);
+            };
+        });
+    }
+
+    onTabbing(event, modalType, findElementToFocus) {
+        const isTAB = event.which === tabKeyCode;
+
+        if (!isTAB) return;
+
+        const $collection = focusableElements[modalType]();
+        const collectionLastIdx = $collection.length - 1
+
+        const $currentElement = $(document.activeElement);
+        const currentElementIdx = $collection.index($currentElement);
+
+        const direction = event.which === tabKeyCode && event.shiftKey ? 'backwards' : 'forwards';
+        
+        /* to jump to the first or last element (depends on direction)
+        if focused element NOT in collection
+        it is possible if user will focus for example element with tabindex=-1 using mouse */
+        const isValidElementActive = currentElementIdx !== -1;
+
+        let startingPoint;
+        if (isValidElementActive) {
+            startingPoint = currentElementIdx
+        } else if (direction === 'forwards') {
+            startingPoint = collectionLastIdx;
+        } else if (direction === 'backwards') {
+            startingPoint = 0;
+        }
```

Comment:
For me, it's look like a separate small function with switch(true){} statement if you want to use string direction var instead of boolean